### ISSUE TITLE
isValidRequest randomly failing

### ISSUE THREAD

**Issue by WhyCantYouCode** [sanity-io/webhook-toolkit#9]
The isValidRequest is randomly failing and returning a false value despite our secret never changing nor our site code, only sanity content.  This issue has arisen out of nowhere and even after making a new API secret, still failing.  We have been successfully using this method for a couple of months prior.

**Comment by msafi** [comment]
I have a similar issue and have spent this morning investigating it. It seems that this library computes a different signature than the one in the webhook request headers in some cases.

For example, in my I have a webhook projection that contains the following:

```json
"quote": "Our 46th president, @joebiden. I’ve been waiting to post this photo because I knew deep down you would be our next President. Today we celebrate the millions of Americans who overcame barriers to the ballot box, mobilized their communities & voted in record breaking numbers."
```

You see that `&` at the end? It caused the library to compute a different signature than the back-end. When I remove the `&`, it works as expected.

**Comment by msafi** [comment]
My workaround:

I removed all data in the projection. I just kept the `_id` and the `delta::operation()`. Then when the webhook request gets through to my API (it validates correctly because there are no `&`s anywhere), I use the Sanity Client with CDN turned off to get the fresh data that was previously being sent through the webhook projection.

**Comment by jordan-mcrae** [comment]
This also happens when expanding references in the projection field. E.g.:

```
{
  ...,
  topics[] ->
```

**Comment by judofyr** [comment]
This is now fixed in 2.0.0. Note that this is a breaking change and requires some changes in how you set it up: https://github.com/sanity-io/webhook-toolkit#usage
